src/org/thoughtcrime/securesms/RegistrationActivity.java

```diff
@@ -458,12 +463,16 @@ private void updateProviderInfo() {
 
     private void maybeCleanProviderInfo() {
         if (provider!=null && providerLayout.getVisibility()==View.VISIBLE) {
-            DcProvider newProvider = getContext(this).getProviderFromEmailWithDns(emailInput.getText().toString());
-            if (newProvider == null
-             || !newProvider.getOverviewPage().equals(provider.getOverviewPage())) {
-                provider = null;
-                providerLayout.setVisibility(View.GONE);
-            }
+            Util.runOnBackground(() -> {
+                DcProvider newProvider = getContext(this).getProviderFromEmailWithDns(emailInput.getText().toString());
+                Util.runOnMain(() -> {
+                    if (newProvider == null
+                     || !newProvider.getOverviewPage().equals(provider.getOverviewPage())) {
+                        provider = null;
+                        providerLayout.setVisibility(View.GONE);
+                    }
+                });
+            });
```

Comment:
as maybeCleanProviderInfo() is called on every keypress: if getProviderFromEmailWithDns() is really slow (what this pr aims to protect against),we'll get a lot of concurrent calls to getProviderFromEmailWithDns(), making things even slower

so, we could debounce or so the calls - however, it also seems sufficient to clear the provider info unconditionally if the user corrects and already entered address.

```suggestion
            provider = null;
            providerLayout.setVisibility(View.GONE);
```